### PR TITLE
fix empty values only workspaces; adding tests

### PR DESCRIPTION
**Pull request by chinook25**

### CODE REVIEW

File: app/ts/Workspace/WorkspaceContextUtil.test.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import IScale from '@shared/interface/IScale';
+import {
+  createCriteriaWithSwappedDataSources,
+  createNewOrdering,
+  hasScalevalues
+} from './WorkspaceContextUtil';
+
+describe('WorkspaceContextUtil', () => {
+  describe('hasScalevalues', () => {
+    it('should return true is there is atleast one value for the scales', () => {
```
Comment (DanielReid) [inline app/ts/Workspace/WorkspaceContextUtil.test.ts:14]: if there is at least

File: app/ts/Workspace/WorkspaceContextUtil.test.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      const scales: Record<string, Record<string, IScale>> = {
+        ds1id: {alt1Id: {'2.5%': 0, '50%': 50, '97.5%': 100, mode: 50}},
+        ds2id: {alt1Id: {'2.5%': null, '50%': null, '97.5%': null, mode: null}}
+      };
+      expect(hasScalevalues(scales)).toBeTruthy();
+    });
+    it('should return false is there is no value for the scales', () => {
```
Comment (DanielReid) [inline app/ts/Workspace/WorkspaceContextUtil.test.ts:21]: if there is no

**Review by kzalite — APPROVED** [review]

File: app/ts/Workspace/WorkspaceContext.tsx [hunk 3]
```diff
@@ -67,20 +74,11 @@ export function WorkspaceContextProviderComponent({
   }, []);
 
   useEffect(() => {
-    if (scales && oldWorkspace) {
+    if (hasScalevalues(scales) && oldWorkspace) {
       setObservedRanges(calculateObservedRanges(scales, workspace));
```
Comment (DanielReid) [inline app/ts/Workspace/WorkspaceContext.tsx:78]: interesting that it checks for oldworkspace but then uses `workspace` - i wonder if there's a race condition here

File: app/ts/Workspace/WorkspaceContextUtil.test.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      const scales: Record<string, Record<string, IScale>> = {
+        ds1id: {alt1Id: {'2.5%': null, '50%': null, '97.5%': null, mode: null}},
+        ds2id: {alt1Id: {'2.5%': null, '50%': null, '97.5%': null, mode: null}}
+      };
+      expect(hasScalevalues(scales)).toBeFalsy();
+    });
+  });
+
+  describe('createNewOrdering', () => {
+    it('should return an ordering for the given alternatives and critria', () => {
```
Comment (DanielReid) [inline app/ts/Workspace/WorkspaceContextUtil.test.ts:31]: criteria

File: app/ts/Workspace/WorkspaceContextUtil.test.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        alternatives: ['alt2Id', 'alt1Id'],
+        criteria: ['crit2Id', 'crit1Id'],
+        dataSources: ['ds3Id', 'ds2Id', 'ds1Id']
+      };
+      expect(result).toEqual(expectedResult);
+    });
+  });
+
+  describe('createCriteriaWithSwappedDataSources', () => {
+    const criteria: ICriterion[] = [
```
Comment (DanielReid) [inline app/ts/Workspace/WorkspaceContextUtil.test.ts:72]: missing `it`

**Review by DanielReid — CHANGES_REQUESTED** [review]
typoes, missing test it

File: app/ts/Workspace/WorkspaceContext.tsx (hunk 3 above)
Comment (chinook25) [reply app/ts/Workspace/WorkspaceContext.tsx:78]: Don't know, haven't looked in to it, as it wasn't part of this bug.

**Review by DanielReid — APPROVED** [review]

**Pull request merged**